src/GameManipulator.php

```diff
@@ -121,6 +121,65 @@ public function isAbandoned(Game $game): bool
         return $game->getStatus() === GameStatusEnum::ABANDONED;
     }
 
+    /**
+     * Tells if one of the player won the game
+     * 
+     * @param Game $game
+     * 
+     * @return Player|null
+     */
+    public function hasPlayerWon(Game $game): ?Player
+    {
+        if ($this->fleetHasSunk($game, $game->getPlayer1())) {
+            $this->setGameAsWon($game);
+            return $game->getPlayer2();
+        } elseif ($this->fleetHasSunk($game, $game->getPlayer2())) {
+            $this->setGameAsWon($game);
+            return $game->getPlayer1();
+        }
+
+        return null;
+    }
+
+    /**
+     * Update the status of the game to OVER
+     * 
+     * @param Game $game
+     * 
+     * @return Game
+     */
+    public function setGameAsWon(Game $game): Game
+    {
+        $game->setStatus(GameStatusEnum::OVER);
+        $this->entityManager->persist($game);
+        $this->entityManager->flush();
+
+        return $game;
+    }
+
+    /**
+     * Used to know if all the ships of a player have sunk
+     * 
+     * @param Game $game
+     * @param Player $player
+     * 
+     * @return bool
+     */
+    public function fleetHasSunk(Game $game, Player $player): bool
+    {
+        $fleetHasSunk = true;
+
+        $ships = $this->shipRepository->getPlayerShips($game, $player);
+        foreach ($ships as $ship) {
+            if (!$this->isShipSunk($ship, $game, $this->getOpponentPlayer($game, $player))) {
+                $fleetHasSunk = false;
```

Comment:
You can return false directly instead of break. And return true at the end. This way you don't need a `$fleetHasSunk` variable.

Or use ArrayCollection->exists to loop in a functionnal way